src/genius/family-form/family-form.component.ts

```diff
@@ -163,35 +242,47 @@ export class FamilyFormComponent {
     this.family.mother = person;
   }
 
-  addChild(person: Person): void {
-    this.family.children.push(person)
+  checkChild(index: number): Boolean {
+    return Object.keys(this.getChild(index)).length > 0
   }
 
-  getFamilyList() {
-    return this.dataProvider.getFamilies();
+  getCompleteChildrenAmount(): number {
+    let amount = 0;
+    if (this.getChildren()) {
+      this.getChildren().forEach(child => {
+        if (Object.keys(child).length > 0) amount++;
+      })
+    }
+
+    return amount;
   }
 
-  getPersonsList() {
-    return this.dataProvider.getPersons();
+  getFamilyList(): void {
+    this.dataProvider.getFamilies().subscribe(httpResponseGetFamilies => this.families = httpResponseGetFamilies,
```

Comment:
rename all such parameters to proper naming. this is not httpResponseGetFamilies, your observable returning list of families, so correct name is families